pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -2111,6 +2111,10 @@ public void updateRates(NamespaceStats nsStats, NamespaceBundleStats bundleStats
         bundleStats.producerCount += producers.size();
         topicStatsStream.startObject(topic);
 
+        if (brokerService.pulsar().getConfig().isEnableReplaceProducerStatsWithTopicStats()) {
+            rateIn.calculateRate();
```

Comment:
It seems like we don't skip/update topicStatsHelper.aggMsgRateIn and aggMsgThroughputIn here.